homeassistant/components/spc.py

```diff
@@ -42,81 +34,52 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
-@asyncio.coroutine
-def async_setup(hass, config):
-    """Set up the SPC platform."""
-    hass.data[DATA_REGISTRY] = SpcRegistry()
+async def async_setup(hass, config):
+    """Set up the SPC component."""
+    from pyspcwebgw import SpcWebGateway
 
-    api = SpcWebGateway(hass,
-                        config[DOMAIN].get(CONF_API_URL),
-                        config[DOMAIN].get(CONF_WS_URL))
+    registry = SpcRegistry()
+    hass.data[DATA_REGISTRY] = registry
 
-    hass.data[DATA_API] = api
+    @callback
```

Comment:
Something should either be declared async or a callback. Callback are methods that are not async but are safe to run inside the eventloop.